### PR TITLE
P11 Dev: Gtest cases for IPZ parser

### PR DESCRIPTION
**Pull request by branupama**
Adding different google unit test cases to verify whether IPZ parser is working for different VPD input.

Test:
Observed all google test cases got passed with different inputs to IPZ parser.

### CODE REVIEW

**Review by PriyangaRamasamy — CHANGES_REQUESTED** [review]
1. Can you write a test case which hits the error "Unable to determine VPD format".
2. Can you get the code coverage report for your gtest cases to get details on number of lines and functions covered and also the ones which are uncovered.

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

File: test/utest_ipz_parser.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+TEST(IpzVpdParserTest, MalformedVpdFile)
+{
+    // Vpd vector size is less than RECORD_MIN(44)
+    nlohmann::json l_json;
+    std::string l_vpdFile("vpd_files/ipz_system_min_record.dat");
+    Parser l_vpdParser(l_vpdFile, l_json);
+
+    EXPECT_THROW(l_vpdParser.parse(), std::exception);
+}
```
[the remark this answers is not in the document]
Comment (branupama) [reply test/utest_ipz_parser.cpp:114]: Empty vector can't be reached to Parser class, it is handled in the parser_factory.cpp itself.

[the remark this answers is not in the document]
Comment (branupama) [reply test/utest_ipz_parser.cpp:114]: Handled in the 'MissingHeader' test case.

[the remark this answers is not in the document]
Comment (branupama) [reply test/utest_ipz_parser.cpp:114]: As we are passing dummy vpdecc.c file for compiling through Jenkins/locally, ECC check will always be passed in this case.
Added test cases 'InvalidRecordOffset' & 'InvalidRecordEccOffset' to check invalid Record offset & Record ECC Offset in the VPD file.

[the remark this answers is not in the document]
Comment (branupama) [reply test/utest_ipz_parser.cpp:114]: Covered in the 'GoodTestCase' test case.

File: test/utest_ipz_parser.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+#include "parser.hpp"
+#include "types.hpp"
+
+#include <cstdint>
+#include <exception>
+#include <fstream>
+
+#include <gtest/gtest.h>
+
+using namespace vpd;
```
[the remark this answers is not in the document]
Comment (branupama) [reply test/utest_ipz_parser.cpp]: Done

File: test/utest_ipz_parser.cpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+TEST(IpzVpdParserTest, GoodTestCase)
+{
+    nlohmann::json l_json;
+    std::string l_vpdFile("vpd_files/ipz_system.dat");
+    Parser l_vpdParser(l_vpdFile, l_json);
+
+    types::IPZVpdMap l_ipzVpdMap =
+        std::get<types::IPZVpdMap>(l_vpdParser.parse());
```
[the remark this answers is not in the document]
Comment (branupama) [reply test/utest_ipz_parser.cpp]: Updated the code.

File: test/utest_ipz_parser.cpp [hunk 4]
```diff
@@ -0,0 +1,77 @@
+#include "exceptions.hpp"
```
[the remark this answers is not in the document]
Comment (branupama) [reply test/utest_ipz_parser.cpp]: Updated the code.

File: test/utest_ipz_parser.cpp [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        if ((l_vpdItr->second).end() != l_kwValItr)
+        {
+            l_description = l_kwValItr->second;
+        }
+    }
+
+    ASSERT_EQ(l_description, "SYSTEM BACKPLANE");
+}
+
+TEST(IpzVpdParserTest, VpdFileNotExists)
```
[the remark this answers is not in the document]
Comment (branupama) [reply test/utest_ipz_parser.cpp]: Updated.

**Review by SunnySrivastava1984 — APPROVED** [review]

**Pull request merged**